test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresUnreferencedCodeCapability.cs

```diff
@@ -163,14 +157,141 @@ static void RequiresUnreferencedCodeOnlyThroughReflection ()
+		static void TestTypeWhichOverridesMethodVirtualMethodRequiresUnreferencedCodeOnBase ()
+		{
+			BaseType tmp = new TypeWhichOverridesMethod ();
+			tmp.VirtualMethodRequiresUnreferencedCode ();
+		}
+
+		class StaticCtor
+		{
+			[RequiresUnreferencedCode ("Message for --TestStaticCtor--")]
+			static StaticCtor ()
+			{
+			}
+		}
+
+		[ExpectedWarning ("IL2026", "--TestStaticCtor--")]
+		static void TestStaticCctorRequiresUnreferencedCode ()
+		{
+			_ = new StaticCtor ();
+		}
+
+		public class DynamicallyAccessedTypeWithRequiresUnreferencedCode
+		{
+			[RequiresUnreferencedCode ("Message for --DynamicallyAccessedTypeWithRequiresUnreferencedCode.RequiresUnreferencedCode--")]
+			public void RequiresUnreferencedCode ()
+			{
+			}
+		}
+
+		static void TestDynamicallyAccessedMembersWithRequiresUnreferencedCode (
+			[DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.PublicMethods)] Type type)
+		{
+		}
+
+		[LogDoesNotContain ("ImplementationClass.RequiresUnreferencedCodeMethod")]
+		[ExpectedWarning ("IL2026", "--IRequiresUnreferencedCode.RequiresUnreferencedCodeMethod--")]
+		static void TestInterfaceMethodWithRequiresUnreferencedCode ()
+		{
+			IRequiresUnreferencedCode inst = new ImplementationClass ();
+			inst.RequiresUnreferencedCodeMethod ();
+		}
+
+		class BaseReturnType { }
+		class DerivedReturnType : BaseReturnType { }
+
+		interface IRequiresUnreferencedCode
+		{
+			[RequiresUnreferencedCode ("Message for --IRequiresUnreferencedCode.RequiresUnreferencedCodeMethod--")]
+			public void RequiresUnreferencedCodeMethod ();
+		}
+
+		class ImplementationClass : IRequiresUnreferencedCode
+		{
+			[RequiresUnreferencedCode ("Message for --ImplementationClass.RequiresUnreferencedCodeMethod--")]
+			public void RequiresUnreferencedCodeMethod ()
+			{
+			}
+		}
+
+		abstract class CovariantReturnBase
+		{
+			[RequiresUnreferencedCode ("Message for --CovariantReturnBase.GetRequiresUnreferencedCode--")]
+			public abstract BaseReturnType GetRequiresUnreferencedCode ();
+		}
+
+		class CovariantReturnDerived : CovariantReturnBase
+		{
+			[RequiresUnreferencedCode ("Message for --CovariantReturnDerived.GetRequiresUnreferencedCode--")]
+			public override DerivedReturnType GetRequiresUnreferencedCode ()
+			{
+				return null;
+			}
+		}
+
+		[LogDoesNotContain ("--CovariantReturnBase.GetRequiresUnreferencedCode--")]
+		[ExpectedWarning ("IL2026", "--CovariantReturnDerived.GetRequiresUnreferencedCode--")]
+		static void TestCovariantReturnCallOnDerived ()
+		{
+			var tmp = new CovariantReturnDerived ();
+			tmp.GetRequiresUnreferencedCode ();
+		}
+
+		[ExpectedWarning ("IL2026", "--Method--")]
+		static void TestRequiresInMethodFromCopiedAssembly ()
+		{
+			var tmp = new RequiresUnreferencedCodeInCopyAssembly ();
+			tmp.Method ();
+		}
+
+		[ExpectedWarning ("IL2026", "--MethodCalledThroughReflection--")]
+		static void TestRequiresThroughReflectionInMethodFromCopiedAssembly ()
```

Comment:
Given the use case in DispatchProxy (that one is about DynamicallyAccessedMember, but it's similar) - can you please add another test case where we apply the `All` annotation to an interface and we validate that it warns if one of the methods on the interface is annotated with `RequiresUnreferencedCode` - maybe a more complex one if the method is on another interface, for example:

```C#
interface A { [RequiresUnreferencedCode] void DoA (); }
interface B : A { void DoB (); }

RequiresAll(typeof(B));
```